lib/hacluster.pm

```diff
@@ -55,6 +57,18 @@ our $crm_mon_cmd     = 'crm_mon -R -r -n -N -1';
 our $softdog_timeout = 60;
 our $prev_console;
 
+sub exec_csync {
+    # Sometimes we need to run csync2 twice to have all the files updated!
+    assert_script_run 'csync2 -vxF ; sleep 2 ; csync2 -vxF';
+}
+
+sub add_file_in_csync {
+    my %args      = @_;
+    my $conf_file = $args{conf_file} // '/etc/csync2/csync2.cfg';
+    assert_script_run "grep -q $args{value} $conf_file || sed -i 's|^}\$|include $args{value};\\n}|' $conf_file";
```

Comment:
I think it would be better to check that the content of `$args{value}` is valid, otherwise there is a risk of the `assert_script_run` failing due to a timeout, or of something invalid being added to `$args{conf_file}`.

Also, would it make sense to check whether the content of `$args{value}` references a valid file?